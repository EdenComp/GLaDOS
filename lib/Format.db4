fn println(str) {
    print(str + "\n");
}

fn btoi(b) {
    if (b) {
        return 1;
    } else {
        return 0;
    }
}

fn itob(i) {
    if (i == 0) {
        return false;
    } else {
        return true;
    }
}
