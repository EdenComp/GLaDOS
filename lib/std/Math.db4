// Package std -- Math

fn is_even(a) {
    return (a % 2) == 0;
}

fn is_neg(a) {
    return a < 0;
}

fn abs(a) {
    if (a < 0) {
        return a * -1;
    }
    return a;
}

fn max(a, b) {
    if (a >= b) {
        return a;
    }
    return b;
}

fn min(a, b) {
    if (a <= b) {
        return a;
    }
    return b;
}

fn clamp(nb, min_nb, max_nb) {
    int low = min(nb, max_nb);

    return max(low, min_nb);
}

fn factorial(x) {
    if (x < 0) {
        return 0;
    } elif (x < 2) {
        return 1;
    } else {
        return x * factorial(x - 1);
    }
}
