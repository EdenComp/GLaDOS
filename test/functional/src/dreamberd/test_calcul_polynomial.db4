import "std:Format";

fn main() {
    fn calculate_polynomial(x) {
        return ((((2*(x**3))) + (3*(x**2))) - ((5*x) + 1));
    }

    fn draw_line(length) {
        for (int j = 0; j < length; j++) {
            print("#");
        }
        println("");
    }

    int a = 2;
    int b = 3;
    int c = -5;
    int d = 1;

    int start = -50;
    int end = 10;

    println("Calcul du polynÃ´me 2x^3 + 3x^2 - 5x + 1 :");

    int result = 0;
    for (int x = start; x <= end; x = x + 1) {
        result = calculate_polynomial(x);
        print("f(" + x + ") = " + result + " : ");

        for (int i = 0; i < result; i = i + 1) {
            if ((i % 2) == 0) {
                print("*");
            }
        }
        println("");
    }

    draw_line(180);

    return 0;
}

main();
