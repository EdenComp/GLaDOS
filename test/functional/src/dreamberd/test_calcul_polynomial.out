Calcul du polyn√¥me 2x^3 + 3x^2 - 5x + 1 :
f(-50) = -242251 : 
f(-49) = -227851 : 
f(-48) = -214033 : 
f(-47) = -200785 : 
f(-46) = -188095 : 
f(-45) = -175951 : 
f(-44) = -164341 : 
f(-43) = -153253 : 
f(-42) = -142675 : 
f(-41) = -132595 : 
f(-40) = -123001 : 
f(-39) = -113881 : 
f(-38) = -105223 : 
f(-37) = -97015 : 
f(-36) = -89245 : 
f(-35) = -81901 : 
f(-34) = -74971 : 
f(-33) = -68443 : 
f(-32) = -62305 : 
f(-31) = -56545 : 
f(-30) = -51151 : 
f(-29) = -46111 : 
f(-28) = -41413 : 
f(-27) = -37045 : 
f(-26) = -32995 : 
f(-25) = -29251 : 
f(-24) = -25801 : 
f(-23) = -22633 : 
f(-22) = -19735 : 
f(-21) = -17095 : 
f(-20) = -14701 : 
f(-19) = -12541 : 
f(-18) = -10603 : 
f(-17) = -8875 : 
f(-16) = -7345 : 
f(-15) = -6001 : 
f(-14) = -4831 : 
f(-13) = -3823 : 
f(-12) = -2965 : 
f(-11) = -2245 : 
f(-10) = -1651 : 
f(-9) = -1171 : 
f(-8) = -793 : 
f(-7) = -505 : 
f(-6) = -295 : 
f(-5) = -151 : 
f(-4) = -61 : 
f(-3) = -13 : 
f(-2) = 5 : ***
f(-1) = 5 : ***
f(0) = -1 : 
f(1) = -1 : 
f(2) = 17 : *********
f(3) = 65 : *********************************
f(4) = 155 : ******************************************************************************
f(5) = 299 : ******************************************************************************************************************************************************
f(6) = 509 : ***************************************************************************************************************************************************************************************************************************************************************
f(7) = 797 : ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
f(8) = 1175 : ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
f(9) = 1655 : ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
f(10) = 2249 : *********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
####################################################################################################################################################################################
